### ISSUE TITLE
SC2142 false positive on positional parameters in alias with escaping

### ISSUE THREAD

**Issue by padde** [koalaman/shellcheck#2635]
#### For bugs
- Rule Id (if any, e.g. SC1000): [SC2142](https://shellcheck.net/wiki/SC2142)
- My shellcheck version (`shellcheck --version` or "online"): `0.8.0`
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Description

When the `$` is escaped with `\$` it should not be considered as a parameter.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

alias a="awk '{print \$0}'"
```

#### Here's what shellcheck currently says:

```
In example.sh line 3:
alias a="awk '{print \$0}'"
      ^-------------------^ SC2142 (error): Aliases can't use positional parameters. Use a function.

For more information:
  https://www.shellcheck.net/wiki/SC2142 -- Aliases can't use positional para...
```

#### Here's what I wanted or expected to see:

No errors.

**Comment by martin-braun** [comment]
It's also a false positive if you have positional arguments in an anonym function within an alias:

```alias foobar='f() { echo "$@"; }; f'```

**Comment by forthrin** [comment]
+1. Wanted to do this piece of magic as an alias, but first `$` must be escaped, and then SC2142 refuses regardless.

```sh
unik() {
  awk '!x[$0]++'
}

alias unik="awk '!x[$0]++'"

```
